### ISSUE TITLE
Work better with the new Analytics.js code?

### ISSUE THREAD

**Issue by ghost** [hybridgroup/gabba#25]
I'm not sure if these are bugs or not, but I'm using the new analytics.js code & Gabba seems to be using the old version, thus reporting some odd numbers.

1) It's reporting every server-side event as a page view, thus incredibly over-inflating the actual number of hits/visits to my site.

2) It's listing the "agent" as "Gabba [version]" and not the browser/city/etc of each visitor. Not a huge deal, but again, greatly waters down the tracking effectiveness of my GA implementation.

3) Custom Variables have been depricated in the new analytics.js code and have been replaced by custom "dimensions" and "metrics". Are there any plans to incorporate those features into Gabba?

All said, I'm loving using Gabba. It makes far more sense to track server-side events on the server instead of attaching behavior to buttons & whatnot on the front-end. 

I'm not sure if sending pull requests would even be valid (since I think Gabba is written against the old tracking style), but I'd be happy to help with a new version since I use it all the time :)

**Comment by inspire22** [comment]
I'd love to see updates regarding this as well...

**Comment by deadprogram** [comment]
Let's do this! Is a 2.0.dev branch called for here?

**Comment by 14113** [comment]
You can try this:

``` ruby
gabba = Gabba::Gabba.new(account_id, domain_name, request.env['HTTP_USER_AGENT'])
gabba.identify_user(cookies[:__utma], cookies[:__utmz])
gabba.ip(request.env["REMOTE_ADDR"])
gabba.referer(request.env['HTTP_REFERER'])
```

**Comment by deadprogram** [comment]
How is that working?

**Comment by deadprogram** [comment]
Meaning is it working ok? I've not tested, so just asking if anyone else knows :cat:

**Comment by 14113** [comment]
For me it's working, when google analytics knows IP address and user agent,

**Comment by deadprogram** [comment]
Aha! Sounds like the API is more restrictive, but still works as before. We need to add this to the documentation then, it seems.

**Comment by mleszcz** [comment]
Google created a special protocol to send server-side events to Universal Google Analytics. Try it out, it looks really simple https://developers.google.com/analytics/devguides/collection/protocol/v1/
Someone already successfully used it https://coderwall.com/p/xpjhbw on production.

**Comment by deadprogram** [comment]
That looks very cool!

**Comment by forrest** [comment]
:+1:

**Comment by jjulian** [comment]
I need to be sure before I deploy this in production: To ensure that gabba works after a Universal Analytics upgrade, the ip and referer should be passed explicitly as noted by @14113. True?

**Comment by daymun** [comment]
Is there a way to identify users with gabba when using analytics.js?
The utm cookies are not used anymore: https://developers.google.com/analytics/devguides/collection/analyticsjs/cookie-usage#analyticsjs

**Comment by toadle** [comment]
I'm also on the problem on not being able the identify the user, since we use Universal analytics. 
Are there any plans to support this? I'd be willing to help!

**Comment by daymun** [comment]
@toadle --
I ended up using the [staccato gem](https://github.com/tpitale/staccato) because it works well with the newer measurement protocol. Might be better suited for your needs as well.

**Comment by toadle** [comment]
@daymun Thx for the advise. I also thought about that. Did you somehow solve the problem of identifying the user in the server side GA call?

**Comment by daymun** [comment]
Yes, take a look at the end of the readme: https://github.com/tpitale/staccato#tracker-hit-defaults
Try: `Staccato.tracker('UA-XXXX-Y', client_id, { user_id: @user.id })`

**Comment by epugh** [comment]
I'm getting notifications about the `_gid` cookie and SameSite issues...    I think generated by this Gem...    Has everyone moved on to a new gem?   I will check out staccato...
